include/rmm/mr/device/pool_memory_resource.hpp

```diff
@@ -130,100 +118,72 @@ class pool_memory_resource final : public device_memory_resource {
    */
   Upstream* get_upstream() const noexcept { return upstream_mr_; }
 
- private:
-  using id_type    = uint32_t;
-  using block      = rmm::mr::detail::block;
-  using free_list  = rmm::mr::detail::free_list<>;
+ protected:
+  using free_list  = detail::coalescing_free_list;
+  using block_type = typename free_list::block_type;
   using lock_guard = std::lock_guard<std::mutex>;
 
+  void initialize_pool(cudaStream_t stream)
+  {
+    cudaDeviceProp props;
+    int device{0};
+    RMM_CUDA_TRY(cudaGetDevice(&device));
+    RMM_CUDA_TRY(cudaGetDeviceProperties(&props, device));
+
+    if (initial_pool_size_ == default_initial_size) {
+      initial_pool_size_ = props.totalGlobalMem / 2;
+    }
+
+    initial_pool_size_ = rmm::detail::align_up(initial_pool_size_, allocation_alignment);
```

Comment:
Not sure if the original or the code below is preferable. Also, I just spent some time trying to figure out if lambda capture list by copy (aka `[=]`) will elide all the copies by default when the lambda isn't marked `mutable`. Now I am just confused on why lambda capture by reference (aka `[&]`) isn't `const` reference, compared to the `const` value. https://www.godbolt.org/z/j8z3sY ¯\_(ツ)_/¯
```cpp
    initial_pool_size_ = [init = initial_pool_size_, &allocation_alignment] { 
        auto const pool_size = init == default_initial_size ? props.totalGlobalMem / 2 : init;
        return rmm::detail::align_up(pool_size, allocation_alignment);
    }();
```

Reply:
How about just 
```suggestion
    initial_pool_size_ =  rmm::detail::align_up((initial_pool_size_ == default_initial_size) ? 
        props.totalGlobalMem / 2 : initial_pool_size_, allocation_alignment);
```
?
Modulo what clang-format decides.

Reply:
That looks pretty good as well :)

Reply:
Done.